Ui/Component/Listing/Column/Data.php

```diff
@@ -0,0 +1,135 @@
+<?php
+declare(strict_types=1);
+
+namespace Rvvup\Payments\Ui\Component\Listing\Column;
+
+use Magento\Framework\App\CacheInterface;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\View\Asset\Repository;
+use Magento\Sales\Api\OrderRepositoryInterface;
+use Magento\Framework\View\Element\UiComponent\ContextInterface;
+use Magento\Framework\View\Element\UiComponentFactory;
+use Magento\Ui\Component\Listing\Columns\Column;
+
+class Data extends Column
+{
+    private const RVVUP_PAYMENTS_IMAGES_GREEN_SVG = 'Rvvup_Payments::images/green.svg';
+    private const RVVUP_PAYMENTS_IMAGES_GREY_SVG = 'Rvvup_Payments::images/grey.svg';
+    private const RVVUP_PAYMENTS_IMAGES_RED_SVG = 'Rvvup_Payments::images/red.svg';
+
+    /** @var OrderRepositoryInterface */
+    private $orderRepository;
+
+    /** @var CacheInterface */
+    private $cache;
+
+    /** @var Repository */
+    private $assetRepo;
+
+    /**
+     * @var RequestInterface
+     */
+    protected $request;
+
+    /**
+     * @param ContextInterface $context
+     * @param UiComponentFactory $uiComponentFactory
+     * @param OrderRepositoryInterface $orderRepository
+     * @param CacheInterface $cache
+     * @param Repository $assetRepo
+     * @param RequestInterface $request
+     * @param array $components
+     * @param array $data
+     */
+    public function __construct(
+        ContextInterface $context,
+        UiComponentFactory $uiComponentFactory,
+        OrderRepositoryInterface $orderRepository,
+        CacheInterface $cache,
+        Repository $assetRepo,
+        RequestInterface $request,
+        array $components = [],
+        array $data = []
+    ) {
+        $this->orderRepository = $orderRepository;
+        $this->cache = $cache;
+        $this->assetRepo = $assetRepo;
+        $this->request = $request;
+        parent::__construct($context, $uiComponentFactory, $components, $data);
+    }
+
+    public function prepareDataSource(array $dataSource): array
+    {
+        if (isset($dataSource['data']['items'])) {
+            foreach ($dataSource['data']['items'] as & $item) {
+                if (isset($item['payment_method'])) {
```

Comment:
can we return early here? I dont think we need to load from cache or do any processing if we know its not cards.

Something like: 
```suggestion
                if (!isset($item['payment_method']) || strpos($item['payment_method'], 'rvvup_CARD') === -1) {
                   continue;
                }
```